custom/icds/location_rationalization/parser.py

```diff
@@ -116,3 +117,40 @@ def validate(self):
                     location.location_id, ",".join(missing_site_codes)
                 ))
 
+    def process(self):
+        self._load_locations()
+        for location_type in list(reversed(self.location_types)):
+            for operation, transitions in self.transitions[location_type].items():
+                self._perform_operation(operation, transitions)
+
+    def _load_locations(self):
```

Comment:
Nitpick: what would you think of making `locations_by_site_code` a memoized property? Functions with names like `load` bug me because they can mean many different things.

Reply:
https://github.com/dimagi/commcare-hq/pull/26898/commits/3157234282d14deef148671d1db17002557a479d